ideascube/configuration/__init__.py

```diff
@@ -40,15 +44,46 @@ def get_config(namespace, key):
         return default_value
 
     if not isinstance(config.value, expected_type):
-        logger.error(
-            'The stored value for %s is of type %s instead of %s. This should'
-            ' never have happened.'
-            % (config, type(config.value), expected_type))
+        logger.error(IMPOSSIBLE_MESSAGE % (
+            config, type(config.value), expected_type))
         return default_value
 
     return config.value
 
 
+def reset_config(namespace, key):
+    """Reset a configuration option to its default value
+
+    Args:
+        namespace (str): The configuration namespace.
+        key (str): The configuration key inside the namespace.
+
+    Raises:
+        ideascube.configuration.exceptions.NoSuchConfigurationNamespaceError:
+            The requested namespace does not exist.
+        ideascube.configuration.exceptions.NoSuchConfigurationKeyError:
+            The requested key does not exist in the requested namespace.
+    """
+    from .models import Configuration
+    from .registry import get_expected_type
+
+    expected_type = get_expected_type(namespace, key)
+
+    try:
+        config = Configuration.objects.get(namespace=namespace, key=key)
+
+    except Configuration.DoesNotExist:
+        # The configuration had never been modified, nothing to reset
+        pass
+
+    else:
+        if not isinstance(config.value, expected_type):
+            logger.error(IMPOSSIBLE_MESSAGE % (
+                config, type(config.value), expected_type))
+
+        config.delete()
+
```

Comment:
I have been never fan of this syntax (try: ... except: ... else: ...)
The `else` always means 'in the other case'. So it means (to me) 'not the normal code path'
With the `try`, `else` means if no exception (so 'the normal code path')
I try to avoid if possible.

In this case for example, the `try` block is here to test if config has been set. At first sight, I tend to think that the else block is done if the config is not set. This is totally opposed of what we do as it is actually the normal behavior.

If we want to do early exit, I prefer this syntax :
```python
try:
    config = Configuration.objects.get(namespace=namespace, key=key)
except Configuration.DoesNotExist:
    # The configuration had never been modified, nothing to reset
    return

if not isinstance(config.value, expected_type):
    logger.error(IMPOSSIBLE_MESSAGE % (
        config, type(config.value), expected_type))

config.delete()
```

It has avantage to make the exit early pattern more visible and do not hide the normal code path behind a `else`
It also remove a level of indentation *

(*) authority argument : This is why Linus Torvald want 8 space indentation in the kernel : Avoid too many indentation.
https://www.kernel.org/doc/Documentation/CodingStyle

> Now, some people will claim that having 8-character indentations makes
the code move too far to the right, and makes it hard to read on a
80-character terminal screen.  The answer to that is that if you need
more than 3 levels of indentation, you're screwed anyway, and should fix
your program.